api/routers/feature.py

```diff
@@ -0,0 +1,205 @@
+from typing import Annotated
+
+import datastore_pb2 as dstore
+import formatters
+from fastapi import APIRouter
+from fastapi import Path
+from fastapi import Query
+from geojson_pydantic import Feature
+from geojson_pydantic import FeatureCollection
+from grpc_getter import get_obs_request
+from grpc_getter import get_spatial_extent
+from shapely import geometry
+from utilities import get_datetime_range
+
+router = APIRouter(prefix="/collections/observations")
+
+
+@router.get(
+    "/items", tags=["Collection items"], response_model=Feature | FeatureCollection, response_model_exclude_none=True
+)
+async def search_timeseries(
+    bbox: Annotated[str | None, Query(example="5.0,52.0,6.0,52.1")] = None,
+    datetime: Annotated[
+        str | None,
+        Query(
+            example="2022-12-31T00:00Z/2023-01-01T00:00Z",
+            description="E-SOH database only contains data from the last 24 hours",
+        ),
+    ] = None,
+    ids: Annotated[str | None, Query(description="Comma separated list of time series ids")] = None,
+    parameter_name: Annotated[str | None, Query(alias="parameter-name", description="E-SOH parameter name")] = None,
+    naming_authority: Annotated[
+        str | None,
+        Query(alias="naming-authority", description="Naming authority that created the data", example="no.met"),
+    ] = None,
+    institution: Annotated[
+        str | None,
+        Query(description="Institution that published the data", example="Norwegian meterological institution"),
+    ] = None,
+    platform: Annotated[
+        str | None, Query(description="Platform ID, WIGOS or WIGOS equivalent.", example="0-20000-0-01492")
+    ] = None,
+    standard_name: Annotated[
+        str | None, Query(alias="standard-name", description="CF 1.9 standard name", example="air_temperature")
+    ] = None,
+    unit: Annotated[str | None, Query(description="Unit of observed physical property", example="degC")] = None,
+    instrument: Annotated[str | None, Query(description="Instrument Id")] = None,
+    level: Annotated[
+        str | None,
+        Query(description="Instruments height above ground or distance below surface, in meters", example=2),
+    ] = None,
+    period: Annotated[
+        str | None, Query(description="Duration of collection period in ISO8601", example="PT10M")
+    ] = None,
+    function: Annotated[
+        str | None, Query(description="Aggregation function used to sample obersved property", example="maximum")
+    ] = None,
+    f: Annotated[
+        formatters.Metadata_Formats, Query(description="Specify return format")
+    ] = formatters.Metadata_Formats.geojson,
```

Comment:
Does this follow the OGC API Feature standard for search, or are we going for this for simplicity?
If yes, could you point me to the relevant section?

Reply:
Section [7.15.5]([https://docs.ogc.org/is/17-069r4/17-069r4.html#_parameters_for_filtering_on_feature_properties) allows for defining parameters for filtering on feature properties. But my understanding of this section might be wrong.

Reply:
Agreed!